crates/sncast/src/starknet_commands/deploy.rs

```diff
@@ -17,12 +17,23 @@ use starknet::providers::jsonrpc::HttpTransport;
 use starknet::signers::LocalWallet;
 use starknet_types_core::felt::Felt;
 
+#[derive(Args, Debug, Clone)]
+#[group(required = true, multiple = false)]
+pub struct ContractIdentifier {
+    /// Class hash of contract to deploy
+    #[arg(short = 'g', long)]
+    pub class_hash: Option<Felt>,
+
+    /// Contract name
+    #[arg(long, conflicts_with = "class_hash")]
+    pub contract_name: Option<String>,
+}
+
```

Comment:
nit: Usage of `id` and `group` seems suitable here:
```suggestion
#[derive(Args, Debug, Clone)]
#[group(id = "contract_identifier", required = true, multiple = false)]
pub struct ContractIdentifier {
    /// Class hash of contract to deploy
    #[arg(short = 'g', long, group = "contract_identifier")]
    pub class_hash: Option<Felt>,

    /// Contract name
    #[arg(long, group = "contract_identifier")]
    pub contract_name: Option<String>,
}

```